### PR TITLE
actions: use github actions

### PR DESCRIPTION
**Pull request by v1v**
It does exactly what it says on the tin

### CODE REVIEW

File: .github/test-reporter.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    workflows:
+      - test
+    types:
+      - completed
+
+jobs:
+  report:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/test-report@gh_actions
```
Comment (v1v) [inline .github/test-reporter.yml:15]:
```suggestion
      - uses: elastic/apm-pipeline-library/.github/actions/test-report@current
```

File: .github/test.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      uses: actions/upload-artifact@v3
+      with:
+        name: test-results
+        path: build/*.xml
+
+  pre-commit:
+    name: Run pre-commit
+    runs-on: ubuntu-latest
+    steps:
+    - uses: elastic/apm-pipeline-library/.github/actions/pre-commit@gh_actions
```
Comment (v1v) [inline .github/test.yml:40]:
```suggestion
    - uses: elastic/apm-pipeline-library/.github/actions/pre-commit@current
```

**Review by amannocci — APPROVED** [review]
LGTM
Note: Test reports will need to be done in another PR.

**Comment by v1v** [comment]
> @v1v This one looks ready to go.

There is a missing piece regarding reporting test results, but as far as I see there are only 13 tests.

I'll remove the scaffolding I created to report tests since the missing part is regarding transforming Go tests to JUnit and https://github.com/elastic/apm-pipeline-library/pull/1946 could help or if we contribute to the upstream https://github.com/dorny/test-reporter/issues/189

**Pull request merged**